test/types/string/methods/testDedent.chpl

```diff
@@ -0,0 +1,89 @@
+/* Temporary dummy method to make sure I can compile this test */
+proc string.dedent(columns=0, ignoreFirst=true) { return this; }
+
+var testStrings = [
+
+  // [0] Common use-case
+  """a
+     b
+     c""".dedent(),
+
+  // [1] Newlines before/after
+  """
+  a
+  b
+  c
+  """.dedent(),
+
+  // [2] Lines with only whitespace are normalized to a single newline.
+  // Note there is trailing whitespace in this example:
+  """  
+  a
+    
```

Comment:
I think I would've expected the whitespace here to be preserved given that no columns are trimmed off for a, b, c.  Interesting variants would be:

```
"""
  a
     // one space, two spaces, four spaces, no spaces
  b
  c
""".dedent()
```

where I'd expect two spaces to be removed when there were two to remove.  Maybe put another way: I don't think that whitespace-only lines should contribute to deciding on the dedent type, but I still think the dedent should be applied to them like any other line.

Reply:
I was following what the python impl. does here, but I don't think it's a big deal to apply the dedentation to the trailing whitespace. It will just be a little more complexity in the implementation.

Reply:
I was imagining the opposite: If the general approach is "eat the leading whitespace as long as it matches the expected pattern until we hit a linefeed or non-whitespace character" then it seems like treating all-whitespace lines is uniform with other cases.  Whereas if the behavior for all-whitespace lines is necessarily "drop everything other than the linefeed" then it seems like a special-case (and one that requires looking at the whole line to identify?)

Reply:
The python implementation does something like this:

(1) Replace any lines containing only white space into `''` (an empty line)
(2) Find the longest sequence of starting space/tabs across all the lines.
(3) Remove that sequence in all lines that begin with that sequence.

So we'll just need to skip (1), and add a check in (2) to make sure we don't check against lines containing only white space. Not a big deal either way.

Reply:
@bradcray - I have switched the tests to this behavior